02-oop-inheritance/script.js

```diff
@@ -0,0 +1,150 @@
+//------------------------OBSERVABLE--------------
+var Observable = function(){
+
+	this.observers = [];
+
+};
+
+
+Observable.prototype.addObserver = function(observer){
+	this.observers.push(observer);
+};
+Observable.prototype.notify = function(event){
+	
+	for(var i = 0; i < this.observers.lenght; i++){
+		this.observers[i].update(event);
+	}
+    for (var i = 0; i < this.observers.length ; i++) {  
+        for (j = 0; j < this.observers[i].eventCallback.length; j++) {
+            
+            if (this.observers[i].eventCallback[j] === event) {
+                if (event === 'play') {
+                    this.observers[i].play(this);
```

Comment:
you could do something to avoid the ifs, try to use the event as key to get the right method to call.